src/SelfGravity/Poi_UserWorkBeforePoisson_GREP.cpp

```diff
@@ -176,81 +242,162 @@ void Mis_UserWorkBeforeNextSubstep_GREP( const int lv, const double TimeNew, con
-   else if (  Mis_CompareRealValue( Time, GREPSgTime[lv][1], NULL, false )  )   Sg = 1;
-   else                                                                         Sg = 1 - GREPSg[lv];
+   if      (  Mis_CompareRealValue( Time, GREP_SgTime[lv][0], NULL, false )  )   Sg = 0;
+   else if (  Mis_CompareRealValue( Time, GREP_SgTime[lv][1], NULL, false )  )   Sg = 1;
+   else                                                                          Sg = 1 - GREP_Sg[lv];
 
 
-// update and combine the spherical-averaged profiles
-   long       TVar         [] = {               _DENS,             _VELR,               _PRES,               _EINT };
-   Profile_t *Prof_Leaf    [] = { DensAve[    lv][Sg], VrAve[    lv][Sg], PresAve[    lv][Sg], EngyAve[    lv][Sg] };
-   Profile_t *Prof_NonLeaf [] = { DensAve[NLEVEL][Sg], VrAve[NLEVEL][Sg], PresAve[NLEVEL][Sg], EngyAve[NLEVEL][Sg] };
+// update the spherical-averaged profiles contributed from non-leaf and leaf patches
+   GREP_Compute_Profile( lv, Sg, PATCH_LEAF    );
+   GREP_Compute_Profile( lv, Sg, PATCH_NONLEAF );
 
 
-   if ( false ) {}
-   /*
-   if ( GREP_ALGO_SWITCH )
-   {
-//    contributions from the leaf patches on level <= lv and the non-leaf patches on level = lv
-      Aux_ComputeProfile( Prof_NonLeaf, GREP_Prof_Center, GREP_Prof_MaxRadius, GREP_Prof_MinBinSize,
-                          GREP_LOGBIN, GREP_LOGBINRATIO, true,  TVar, 4,  0, lv, PATCH_LEAF_PLUS_MAXNONLEAF, Time );
-   }
-   */
+// combine the spherical-averaged profiles
+   GREP_Combine_Profile( GREP_DensAve, lv, Sg, Time, true );
+   GREP_Combine_Profile( GREP_VrAve,   lv, Sg, Time, true );
+   GREP_Combine_Profile( GREP_PresAve, lv, Sg, Time, true );
+   GREP_Combine_Profile( GREP_EngyAve, lv, Sg, Time, true );
+
+
+// compute the pressure if GREP_OPT_PRES == GREP_PRES_BINDATA
+   Profile_t *Dens_Tot = GREP_DensAve[NLEVEL][Sg];
+   Profile_t *Vr_Tot   = GREP_VrAve  [NLEVEL][Sg];
+   Profile_t *Pres_Tot = GREP_PresAve[NLEVEL][Sg];
+   Profile_t *Engy_Tot = GREP_EngyAve[NLEVEL][Sg];
 
-   else
+   if ( GREP_OPT_PRES == GREP_PRES_BINDATA )
    {
-//    retain the empty bins to avoid inconsistent leaf- and non-leaft profiles during combining the profiles
+#     ifdef YE
+      real Passive[ NCOMP_TOTAL - NCOMP_FLUID ] = { 0.0 };
 
-//    contributions from the leaf patches on level = lv
-      Aux_ComputeProfile( Prof_Leaf,    GREP_Prof_Center, GREP_Prof_MaxRadius, GREP_Prof_MinBinSize,
-                          GREP_LOGBIN, GREP_LOGBINRATIO, false, TVar, 4, lv, lv, PATCH_LEAF,    -1.0 );
+      for (int b=0; b<Dens_Tot->NBin; b++)
+      {
+         if ( Dens_Tot->NCell[b] == 0 )   continue;
 
-//    contributions from the non-leaf patches on level = lv
-      Aux_ComputeProfile( Prof_NonLeaf, GREP_Prof_Center, GREP_Prof_MaxRadius, GREP_Prof_MinBinSize,
-                          GREP_LOGBIN, GREP_LOGBINRATIO, false, TVar, 4, lv, lv, PATCH_NONLEAF, -1.0 );
+         Passive[ YE - NCOMP_FLUID ] = Pres_Tot->Data[b];
```

Comment:
Add a comment that Ye has been temporarily stored in the pressure profile.

Reply:
Added